test/tests.js

```diff
@@ -0,0 +1,16 @@
+var simulateClick = function(el){
```

Comment:
I still don't like it. Unit-tests are supposed to be granular and to find functions, not the whole use cases. At the same time, you should not expose functions to the global scope in the final build.

**Think interfaces**. For instance, let's think about possible public functions of the popup blocker module. 

Here is an example.

This module contains all the popup blocking logic and you can easily test it.
```
/**
 * Module that contains all the popup blocking logic.
 */
var popupBlocker = (function() {

    // Expose public API
    return {
        /**
  		 * Returns a window.open wrapper
  		 */
        verifyWindowOpen: function(...) {...},

        /**
  		 * Returns an EventTarget.prototype.dispatchEvent wrapper
  		 */
        verifyDispatchEvent: function(...) {...},

        /**
         * Returns a HTMLElement.prototype.click wrapper
         */
        verifyClick: function(...) {...}
    }
})();
```

A test may look like this:
```
var opened = false;
var assertFunction = function() {
    opened = true;
}
var windowOpen = popupBlocker.verifyWindowOpen(assertFunction);
// Mock the event
window.event = new MouseEvent(123123);
// Run the test
var popup = windowOpen('https://example.org/');
assert.notOk(opened);
assert.ok(popup);
assert.ok(popup.someprop)
```

Then goes iframes initialization:
```
/**
 * Popup blocker initialization
 */ 
var initPopupBlocker = function(window) {

	window.open = popupBlocker.verifyWindowOpen(window.open);
	EventTarget.prototype.dispatchEvent = popupBlocker.verifyDispatchEvent(EventTarget.prototype.dispatchEvent);
	HTMLElement.prototype.click = popupBlocker.verifyClick(HTMLElement.prototype.click);

	initIframes(window);
};
```

Passing code down to the iframes will require a bit of a concatenation, but it does not look hard anyway:
```
var popupBlockerCode = "(function() {" + popupBlocker.toString() + " " + initPopupBlocker.toString() + "..."
```